packages/frontend/src/retro/hooks/useTimer.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useRef, useState } from "react";
+import { useRetroContext } from "../context/RetroContext";
+import { TimerStatus } from "../types/retroTypes";
+
+interface useTimerProps {
+  onTimerFinish: () => void;
+}
+
+export function useTimer({ onTimerFinish }: useTimerProps) {
+  const { timerStatus, timerDuration } = useRetroContext().retroState;
+  const [timeRunning, setTimeRunning] = useState(0);
+  const intervalRef = useRef<NodeJS.Timer>();
+  const remainingTime = timerDuration - timeRunning;
+  const minutes = Math.floor(remainingTime / 1000 / 60);
+  const seconds = Math.round((remainingTime / 1000) % 60);
+  const labelOptions: Intl.DateTimeFormatOptions = {
+    minute: "numeric",
+    second: "numeric",
+    hourCycle: "h23",
+    hour: minutes > 59 ? "numeric" : undefined,
+  };
```

Comment:
```suggestion
  const { timerStatus, timerDuration } = useRetroContext().retroState;
  const [timeRunning, setTimeRunning] = useState(0);
  const intervalRef = useRef<NodeJS.Timer>();
  
  const remainingTime = timerDuration - timeRunning;
  const minutes = Math.floor(remainingTime / 1000 / 60);
  const seconds = Math.round((remainingTime / 1000) % 60);
  
  const labelOptions: Intl.DateTimeFormatOptions = {
    minute: "numeric",
    second: "numeric",
    hourCycle: "h23",
    hour: minutes > 59 ? "numeric" : undefined,
  };
```

I'd create blocks like this where the first block contains actual react state/hooks etc and the rest are just calculations on that.